### PR TITLE
Solid principled follwed initially

### PR DESCRIPTION
**Pull request by gids-dadi**
In this PR the code is refactored so the role of main.rb is only to start the code and ask the user for the initial choice of options.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -20,7 +21,7 @@ def inputs(int)
     end
```
Comment (ShahierNashaat) [inline main.rb]: - [ ] Kindly remove the `starting` word at line 20 in the `inputs` method as it causing an error.

File: main.rb [hunk 2]
```diff
@@ -20,7 +21,7 @@ def inputs(int)
...
   end
 
-  def starting
+  def start
     puts 'Please choose an option by entering a number'
     puts '1 - List all books'
     puts '2 - List all people'
```
Comment (ShahierNashaat) [inline main.rb]: - [ ] Kindly notice that the program has to continue until the user chooses to close it for that you need to repeat showing the options to the user after every operation.

```ruby
def start
    num = 0
    while(num != 7)
      puts 'Please choose an option by entering a number'
      puts '1 - List all books'
      puts '2 - List all people'
      puts '3 - Create a person'
      puts '4 - Create a book'
      puts '5 - Create a rental'
      puts '6 - List all rentals for a given id'
      puts '7 - Exit'
      num = gets.chomp
      inputs(num)
    end
end
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @gids-dadi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
- [ ] Please write the name of `person class` correctly when you require it in the `student class` and `teacher class`.
```ruby
require_relative 'person_class'
```

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @gids-dadi, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**